### PR TITLE
Add a glossary of terms

### PR DESCRIPTION
**Pull request by zeke**
:book:

This PR adds a document that defines some common terms from Electron land.

I'm sure there are many more terms to add, but this is a start.

Feedback welcome!

### CODE REVIEW

File: docs/glossary.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+The renderer process is a browser window in your app. Unlike the main process,
+there can be multiple of these and each is run in a separate process.
+They can also be hidden.
+
+In normal browsers, web pages usually run in a sandboxed environment and are not
+allowed access to native resources. Electron users, however, have the power to
+use Node.js APIs in web pages allowing lower level operating system
+interactions.
+
+### Squirrel
```
Comment (MarshallOfSound) [inline docs/glossary.md]: Might be worth noting that there are two different implementation of Squirrel.

`Squirrel.Mac` and `Squirrel.Windows`

Comment (zeke) [inline docs/glossary.md]: That's covered in the linked autoUpdater docs: http://electron.atom.io/docs/api/auto-updater/#platform-notices

Seems like TMI in this context to me.

File: docs/glossary.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+IPC stands for Inter-Process Communication. Electron uses IPC to send
+serialized JSON messages between the [main] and [renderer] processes.
+
+### libchromiumcontent
+
+A single, shared library that includes the Chromium Content module and all its
+dependencies (e.g., Blink, [V8], etc.).
+
+### main process
```
Comment (zeke) [inline docs/glossary.md]: Good idea, however the terms are in alphabetical order. As an alternative I have added some "See also" links to each process entry.

Comment (YurySolovyov) [inline docs/glossary.md]: Does alphabetical order really worth it here?

Comment (zeke) [inline docs/glossary.md]: I've never seen a glossary of terms that wasn't alphabetized, but I guess you've got a point. In a world where folks can <kbd>CmdOrCtrl</kbd><kbd>f</kbd>, that order may not matter.

Comment (YurySolovyov) [inline docs/glossary.md]: I'm not like super strong about it, just trying to group related stuff together.

**Pull request merged**